libraries/AP_Scripting/examples/aileron_yaw_mix.lua

```diff
@@ -0,0 +1,197 @@
+      _log_data = { target, current, FF, P, I, ret }
+
+      _pid_out.FF = FF
+      _pid_out.P = P
+      _pid_out.I = I
+      _pid_out.target = target
+      _pid_out.actual = current
+
+      return ret
+   end
+
+   -- log the controller internals
+   function self.log(name)
+      logger.write(name,'Targ,Curr,FF,P,I,Total','ffffff',table.unpack(_log_data))
+   end
+
+   function self.send_telemetry()
+      gcs:send_named_float(string.format('YAW_FF'), _pid_out.FF)
+      gcs:send_named_float(string.format('YAW_P'), _pid_out.P)
+      gcs:send_named_float(string.format('YAW_I'), _pid_out.I)
+      gcs:send_named_float(string.format('YAW_TG'), math.deg(_pid_out.target))
+      gcs:send_named_float(string.format('YAW_AC'), math.deg(_pid_out.actual))
+   end
+
+   -- return the instance
+   return self
+end
+
+local yaw_pid = PIFF(AMIX_YAW_FF, AMIX_YAW_P, AMIX_YAW_I, AMIX_YAW_IMAX)
+local last_telem_send = 0.0
+
+function update()
+   -- get the current tilt
+   local tilt_scaled = SRV_Channels:get_output_scaled(k_motor_tilt)
+   if not tilt_scaled then
+      return
+   end
+
+   -- get the current body frame yaw rate
+   local ygyro_rads = ahrs:get_gyro():z()
+   local roll_trate, pitch_trate, yaw_trate = AC_AttitudeControl:get_rpy_rate()
+   local yaw_pid_out = yaw_pid.update(yaw_trate, ygyro_rads)
+
+   -- tilt from 0 to 1
+   local tilt = tilt_scaled * 0.001
+
+   -- get fixed wing aileron
+   local aileron_scaled = SRV_Channels:get_output_scaled(k_aileron)
+   if not aileron_scaled then
+      gcs:send_text(0,string.format("no aileron function", idx))
+      return
+   end
+
+   -- aileron from -1 to 1
+   local aileron = aileron_scaled / 4500.0
+
+   -- get VTOL yaw outputs, sum of FF, P, I and D
+   local yaw_total = yaw_pid_out
+
+   -- tilt angle in radians, 0 means VTOL mode, 1 means fixed wing fwd
+   local tilt_angle_rad = math.pi * 0.5 * tilt
+
+   -- output surface from -1 to 1
+   -- note minus sign on yaw for the frame change
+   local surface = constrain(math.cos(tilt_angle_rad) * (-yaw_total) + math.sin(tilt_angle_rad) * aileron, -1, 1)
+
+   SRV_Channels:set_angle(k_scripting1, 4500)
+   SRV_Channels:set_output_scaled(k_scripting1, surface * 4500)
+
+   yaw_pid.log("AMIX")
+
+   local now = millis():tofloat() * 0.001
+   local dt = now - last_telem_send
+   if dt >= (1.0 / AMIX_TELEM_RATE:get()) then
+      last_telem_send = now
+      yaw_pid.send_telemetry()
+   end
+
+   return update, 1000/UPDATE_RATE_HZ
+end
+
+gcs:send_text(0,string.format("AMIX Loaded"))
+
+-- wrapper around update(). This calls update() and if update faults
+-- then an error is displayed, but the script is not stopped
+function protected_wrapper()
+  local success, err = pcall(update)
+  if not success then
+     gcs:send_text(MAV_SEVERITY_ERROR, "Internal Error: " .. err)
+     -- when we fault we run the update function again after 1s, slowing it
+     -- down a bit so we don't flood the console with errors
+     --return protected_wrapper, 1000
```

Comment:
This line is commented out, so it looks like the code probably doesn't do what the comment says. Is that intentional?